### PR TITLE
libhttpd: update to 2.0

### PR DESCRIPTION
**Pull request by barracuda156**
#### Description

Update to @2.0 + corrections.

###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 10.6.8
Xcode 3.2.6

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?

### CODE REVIEW

File: devel/libhttpd/Portfile [hunk 1]
```diff
@@ -1,32 +1,37 @@
...
+master_sites	    ${homepage}downloads/${name}/
+distname            ${name}-${version}
+checksums           md5     0dbfb1218296c22ffa8c2731f667b967 \
+                    rmd160  2b841b9f5d23edd96704a0787f4a704bf12061c4 \
+                    sha256  a4a9c6c6db30c521e3082950b049ed2f09035f90678ed7052bf856688114f04e \
+                    size    68217
 
-patch      {
-	reinplace "s|/usr/local|${destroot}${prefix}|g" \
+post-extract {
```
Comment (reneeotten) [inline devel/libhttpd/Portfile:27]: this should happen in a "post-patch" phase

File: devel/libhttpd/Portfile [hunk 2]
```diff
@@ -1,32 +1,37 @@
-PortSystem 1.0
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
```
Comment (reneeotten) [inline devel/libhttpd/Portfile:1]: if you add a modeline you need to make sure you conform to it (i.e., no tabs, indent uniformly with spaces as multiples of four). The `lint` warnings in the CI tell you this and you should do `port lint` locally as stated in the PR checklist. I'll repeat it here again, the checklist is not optional - at least do the lint and build it locally before submitting a PR.

File: devel/libhttpd/Portfile [hunk 3]
```diff
@@ -1,32 +1,37 @@
...
+			        features to simplify the creation of web based \
+			        application interfaces.
 
-homepage	http://www.hughes.com.au/products/libhttpd/
-master_sites	${homepage}
-checksums	md5 ea0fb523fdb467702d34185be55e940c
+homepage	        https://hughestech.com.au/
+master_sites	    ${homepage}downloads/${name}/
+distname            ${name}-${version}
+checksums           md5     0dbfb1218296c22ffa8c2731f667b967 \
```
Comment (reneeotten) [inline devel/libhttpd/Portfile:22]: the `md5` checksum is obsolete and can be removed

**Comment by reneeotten** [comment]
thanks for the update @barracuda156 - a few comments above. Please do take comments in other PRs into consideration moving forward when submitting new PRs. I am happy to review things and try and help you out, but would rather not repeat myself unnecessarily many times. 

For example, whitespace changes should be made separately from functional changes (i.e., same PR but different commits), the commit message should be short (I would only list the update here, without typo and the `@` that isn't part of the version). If you really want you could add additional necessary explanations in the body of the message (but then probably something like "add/conform to modeline" instead of "portfile syntax"; you there wasn't anything wrong with the syntax and you didn't change it either).

File: devel/libhttpd/Portfile (hunk 2 above)
Comment (barracuda156) [reply devel/libhttpd/Portfile:1]: Building locally is always done before submission of anything, of course. I will correct other things now.

File: devel/libhttpd/Portfile (hunk 3 above)
Comment (barracuda156) [reply devel/libhttpd/Portfile:22]: Macports suggested checksums line, I copied that. Will remove it.

Comment (reneeotten) [reply devel/libhttpd/Portfile:22]: yeah, that's true: MacPorts will give you the udpated `md5` checksum if that was already present in the Portfile. The `md5` is pretty old and not actively used anymore, but might indeed still be present for some ports that haven't been updated lately - the suggestion is to remove it and only keep the three other ones.

**Comment by barracuda156** [comment]
> thanks for the update @barracuda156 - a few comments above. Please do take comments in other PRs into consideration moving forward when submitting new PRs. I am happy to review things and try and help you out, but would rather not repeat myself unnecessarily many times.
> 
> For example, whitespace changes should be made separately from functional changes (i.e., same PR but different commits), the commit message should be short (I would only list the update here, without typo and the `@` that isn't part of the version). If you really want you could add additional necessary explanations in the body of the message (but then probably something like "add/conform to modeline" instead of "portfile syntax"; you there wasn't anything wrong with the syntax and you didn't change it either).

@reneeotten Thank you. (I will keep in mind to do style & functional changes in separate commits onwards.)

**Pull request merged**